go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1784,15 +1797,15 @@ func flowsForDefaultBridge(bridge *bridgeConfiguration, extraIPs []net.IP) ([]st
 					klog.Infof("Unable to determine IPV4 UDN subnet for the provided family isIPV6: %v", err)
 					continue
 				}
-
-				// Use the filtered subnets for the flow compute instead of the masqueradeIP
-				srcIPOrSubnet = matchingIPFamilySubnet.String()
+				// In addition to the masqIP based flows, we also need the podsubnet based flows for
+				// advertised networks since UDN pod to clusterIP is unSNATed and we need this traffic to be taken into
```

Comment:
UDN pod to clusterIP is unSNATed? Are we talking specifically about KAPI/DNS traffic?

Reply:
Disregard. I understand now this is not SNATed due to advertisement, this is ClusterIP traffic for which a service does not exist on the network and we are just doing what it would be done were the traffic SNATed to the masqIP, which in any case is a dead end for the traffic. This extra detail might be worth adding into the comment.

Reply:
yup exactly what you said but I will make my comments more detailed and easier to understand